impl/src/main/java/org/ehcache/internal/store/disk/Segment.java

```diff
@@ -821,16 +825,24 @@ boolean cleanUpFailedMarker(final K key, final int hash) {
    *
    * @param key     the key
    * @param hash    they hash
-   * @param element the expected element
+   * @param valueHolder   the expected value holder
    * @return true if succeeded
    */
-  boolean flush(final K key, final int hash, final DiskStorageFactory.Element<K, V> element) {
+  boolean flush(final K key, final int hash, DiskValueHolder<V> valueHolder, CachingTier<K, V> cachingTier) {
```

Comment:
I do not think it needs to be a `DiskValueHolder` which would solve the blind cast mentioned earlier.